### PR TITLE
Get node list from nodes

### PR DESCRIPTION
**Pull request by robinhood23rus**
Added function to get nodes from server.
Add doc for multi module.
Code review required.

### CODE REVIEW

File: multi/config_m2.lua [hunk 1]
```diff
@@ -1,9 +1,13 @@
+nodes_load = require("config_load_nodes")
+
 box.cfg {
     listen = 3014,
-    wal_dir = 'm2/xlog',
-    snap_dir = 'm2/snap',
+    wal_dir = 'm2',
+    snap_dir = 'm2',
```
Comment (funny-falcon) [inline multi/config_m2.lua]: Excuse me, was this change important for this PR?

File: multi/multi.go [hunk 2]
```diff
@@ -115,9 +136,19 @@ func (connMulti *ConnectionMulti) deleteConnectionFromPool(addr string) {
 	delete(connMulti.pool, addr)
 }
 
+func (connMulti *ConnectionMulti) clearConnectionPool() {
+	connMulti.mutex.Lock()
+	defer connMulti.mutex.Unlock()
+	for k := range connMulti.pool {
+		delete(connMulti.pool, k)
```
Comment (funny-falcon) [inline multi/multi.go]: Who will close connections?

File: multi/multi.go [hunk 3]
```diff
@@ -156,6 +187,36 @@ func (connMulti *ConnectionMulti) checker() {
...
+				data := resp.Data[0].([]interface{})
+				addrs := make([]string, len(data))
+				for i, v := range data {
+					addrs[i] = v.(string)
+				}
+				connMulti.addrs = addrs
+				connMulti.clearConnectionPool()
+			}
+
+			somebodyAlive, _ := connMulti.warmUp()
```
Comment (funny-falcon) [inline multi/multi.go]: So, you really want to drop all connections and reconnect every 60 seconds?

File: multi/multi.go [hunk 4]
```diff
@@ -115,9 +136,19 @@ func (connMulti *ConnectionMulti) deleteConnectionFromPool(addr string) {
...
+	for k := range connMulti.pool {
+		delete(connMulti.pool, k)
+	}
+}
+
 func (connMulti *ConnectionMulti) checker() {
+
 	for connMulti.getState() != connClosed {
-		timer := time.NewTimer(connMulti.opts.CheckTimeout)
+		checkTimer := time.NewTimer(connMulti.opts.CheckTimeout)
```
Comment (funny-falcon) [inline multi/multi.go]: was variable name important for this PR?

File: multi/config_m2.lua (hunk 1 above)
Comment (robinhood23rus) [reply multi/config_m2.lua]: Sorry, missed it. Cleaned from commit.

File: multi/multi.go (hunk 2 above)
Comment (robinhood23rus) [reply multi/multi.go]: Reworked.

File: multi/multi.go (hunk 3 above)
Comment (robinhood23rus) [reply multi/multi.go]: Reworked to keep actual connections in pool and clean only obsolete

File: multi/multi.go (hunk 4 above)
Comment (robinhood23rus) [reply multi/multi.go]: Sorry, missed it. Cleaned from commit.

File: multi/multi.go [hunk 5]
```diff
@@ -34,38 +51,51 @@ type ConnectionMulti struct {
...
+		control:  make(chan struct{}),
+		pool:     make(map[string]*tarantool.Connection),
 	}
 	somebodyAlive, _ := connMulti.warmUp()
 	if !somebodyAlive {
 		connMulti.Close()
 		return nil, ErrNoConnection
 	}
 	go connMulti.checker()
+	go connMulti.refreshNodes()
```
Comment (funny-falcon) [inline multi/multi.go]: `if NodesGetFunctionName != "" { ... }`

File: multi/multi.go [hunk 6]
```diff
@@ -116,8 +146,10 @@ func (connMulti *ConnectionMulti) deleteConnectionFromPool(addr string) {
 }
 
 func (connMulti *ConnectionMulti) checker() {
+
```
Comment (funny-falcon) [inline multi/multi.go]: excess whitespace

File: multi/multi.go [hunk 7]
```diff
@@ -156,6 +188,58 @@ func (connMulti *ConnectionMulti) checker() {
...
 
+func (connMulti *ConnectionMulti) refreshNodes() {
+	for connMulti.getState() != connClosed {
+		refreshTimer := time.NewTimer(connMulti.opts.ClusterDiscoveryTime)
+		select {
+		case <-refreshTimer.C:
+			if connMulti.getState() == connClosed || len(connMulti.opts.NodesGetFunctionName) <= 0 {
+				continue
+			}
+			resp, err := connMulti.Call(connMulti.opts.NodesGetFunctionName, []interface{}{})
```
Comment (funny-falcon) [inline multi/multi.go]: May be it is better to use `Call17` here?

File: multi/multi.go [hunk 8]
```diff
@@ -172,10 +256,12 @@ func (connMulti *ConnectionMulti) getCurrentConnection() *tarantool.Connection {
 	return connMulti.fallback
 }
 
+// ConnectedNow check for current connections is established
```
Comment (funny-falcon) [inline multi/multi.go]: Send comments with separate PR.

File: multi/multi.go [hunk 9]
```diff
@@ -156,6 +188,58 @@ func (connMulti *ConnectionMulti) checker() {
...
+						con, ok := connMulti.getConnectionFromPool(v)
+						if con != nil && ok {
+							somebodyAlive = true
+						}
+					}
+				}
+				connMulti.addrs = addrs
+
+				if !somebodyAlive {
+					connMulti.Close()
```
Comment (funny-falcon) [inline multi/multi.go]: I doubt it is good idea.

File: multi/multi.go [hunk 10]
```diff
@@ -156,6 +188,58 @@ func (connMulti *ConnectionMulti) checker() {
 	}
 }
 
+func (connMulti *ConnectionMulti) refreshNodes() {
```
Comment (funny-falcon) [inline multi/multi.go]: There is already `checker` loop. Isn't it better to integrate to it?

File: multi/multi.go [hunk 11]
```diff
@@ -117,7 +132,9 @@ func (connMulti *ConnectionMulti) deleteConnectionFromPool(addr string) {
 
 func (connMulti *ConnectionMulti) checker() {
 	for connMulti.getState() != connClosed {
+		refreshTimer := time.NewTimer(connMulti.opts.ClusterDiscoveryTime)
```
Comment (funny-falcon) [inline multi/multi.go]: Блин, нельзя создавать два таймера на каждую итерацию цикла. Будет работать только один.

Думаю, можно переделать на тикеры, создаваемые до цикла.
